prinz_util/src/main/scala/pl/touk/nussknacker/prinz/util/http/RestJsonClient.scala

```diff
@@ -1,27 +1,28 @@
 package pl.touk.nussknacker.prinz.util.http
 
+import io.circe
 import io.circe.{Decoder, Encoder}
 import pl.touk.nussknacker.prinz.util.http.AbstractRestJsonClient.RestClientResponse
-import sttp.client3.{HttpURLConnectionBackend, Identity, SttpBackend}
+import sttp.client3.{HttpURLConnectionBackend, Identity, RequestT, ResponseException, SttpBackend}
 
 class RestJsonClient(baseUrl: String, private val backend: SttpBackend[Identity, Any] = HttpURLConnectionBackend())
   extends AbstractRestJsonClient(baseUrl) {
 
   def postJsonBody[BODY, RESPONSE: Manifest](body: BODY, relativePath: String = "", params: RestRequestParams = EmptyRestRequestParams)
                                             (implicit encoder: Encoder[BODY], decoder: Decoder[RESPONSE]): RestClientResponse[RESPONSE] = {
     val request = createPostJsonRequest(relativePath, body, params)
-    wrapCaughtException(
-      () => request.send(backend).body.left.map(clientExceptionFromResponse),
-      e => Left(RestClientException(e.getMessage))
-    )
+    wrap(request)
   }
 
   def getJson[RESPONSE: Manifest](relativePath: String = "", params: RestRequestParams = EmptyRestRequestParams)
                                  (implicit decoder: Decoder[RESPONSE]): RestClientResponse[RESPONSE] = {
     val request = createGetRequest(relativePath, params)
-    wrapCaughtException(
-      () => request.send(backend).body.left.map(clientExceptionFromResponse),
-      e => Left(RestClientException(e.getMessage))
-    )
+    wrap(request)
   }
+
+  private def wrap[RESPONSE: Manifest]
```

Comment:
imo should be merge in AbstractRestJsonClient if possible with wrapCaughtException because now it looks like code duplication  (it was actually but now it is a separate method)